### ISSUE TITLE
speeding up jekyll build (at least for heroku)

### ISSUE THREAD

**Issue by martenson** [galaxyproject/training-material#1602]
Locally a build of this site takes ~90s.

Heroku has 60s limit on boot time, it retries once and jekyll sometimes runs faster the second time. Here is a profile of a standard build:

```
bundle exec jekyll build --profile
Configuration file: /Users/marten/devel/git/training-material/_config.yml
            Source: .
       Destination: ./_site
 Incremental build: disabled. Enable with --incremental
      Generating...
       Jekyll Feed: Generating feed for posts

Filename                                                                          | Count |    Bytes |   Time
----------------------------------------------------------------------------------+-------+----------+-------
_layouts/tutorial_hands_on.html                                                   |   126 | 7461.86K | 17.200
topics/metagenomics/tutorials/mothur-miseq-sop/content.md                         |     2 |  171.31K |  9.172
topics/assembly/tutorials/unicycler-assembly/tutorial.md                          |     1 |   23.33K |  8.304
topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.md           |     1 |   36.55K |  8.256
topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.md                  |     1 |   78.55K |  5.332
topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.md                        |     1 |   92.76K |  3.854
topics/metagenomics/tutorials/mothur-miseq-sop/background_reference_taxonomies.md |     2 |    5.30K |  3.555
topics/genome-annotation/tutorials/tnseq/tutorial.md                              |     1 |   58.53K |  3.240
topics/epigenetics/tutorials/atac-seq/tutorial.md                                 |     1 |   54.73K |  2.308
topics/metabolomics/tutorials/lcms/tutorial.md                                    |     1 |   77.29K |  2.044
topics/assembly/tutorials/debruijn-graph-assembly/tutorial.md                     |     1 |   16.85K |  1.762
topics/proteomics/tutorials/proteome_annotation/tutorial.md                       |     1 |   23.83K |  1.447
topics/ecology/tutorials/species-distribution-modeling/tutorial.md                |     1 |   19.11K |  1.281
_layouts/base_slides.html                                                         |    65 | 1140.01K |  1.237
topics/contributing/tutorials/create-new-tutorial-content/tutorial.md             |     1 |   25.83K |  1.066
topics/galaxy-data-manipulation/tutorials/name-tags/tutorial.md                   |     1 |   10.91K |  1.064
_layouts/topic.html                                                               |    20 | 1407.29K |  1.017
topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md      |     1 |   67.78K |  0.865
snippets/display_extra_training.md                                                |   104 |  172.75K |  0.854
topics/statistics/tutorials/iwtomics/tutorial.md                                  |     1 |    7.84K |  0.673
_includes/tutorial_list.html                                                      |    20 | 1229.00K |  0.640
topics/admin/tutorials/monitoring/tutorial.md                                     |     1 |   35.27K |  0.628
topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.md        |     1 |   21.36K |  0.628
topics/metagenomics/tutorials/mothur-miseq-sop/background_rarefaction.md          |     2 |    3.62K |  0.597
_layouts/base.html                                                                |   158 | 9675.43K |  0.581
topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.md |     1 |   17.47K |  0.568
_includes/material.jsonld                                                         |   211 | 1744.86K |  0.547
topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.md                |     1 |   41.21K |  0.491
topics/epigenetics/tutorials/methylation-seq/tutorial.md                          |     1 |   13.02K |  0.458
_includes/contributor-badge-list.html                                             |   211 |  125.26K |  0.410
topics/genome-annotation/tutorials/annotation-with-maker/tutorial.md              |     1 |   43.44K |  0.397
topics/epigenetics/tutorials/ewas-suite/tutorial.md                               |     1 |   19.80K |  0.393
topics/proteomics/tutorials/biomarker_selection/tutorial.md                       |     1 |   22.79K |  0.378
_layouts/tutorial_slides.html                                                     |    37 |  436.46K |  0.257
topics/dev/tutorials/data-source-integration/tutorial.md                          |     1 |    1.42K |  0.254
snippets/display_extra_training_slides.md                                         |    26 |   13.92K |  0.170
topics/transcriptomics/tutorials/scrna-raceid/tutorial.md                         |     1 |   72.91K |  0.165
_layouts/hall-of-fame.html                                                        |     1 |  167.19K |  0.136
_includes/contributor-badge.html                                                  |   211 |  123.41K |  0.129
_layouts/home.html                                                                |     1 |  111.65K |  0.107
_includes/contributor-text-list.html                                              |   126 |    4.93K |  0.101
snippets/import_via_link.md                                                       |    52 |   40.36K |  0.095
topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md             |     1 |   35.49K |  0.091
_includes/resource-handson.html                                                   |    20 |  238.95K |  0.090
_includes/instance-dropdown.html                                                  |    47 |   31.55K |  0.088
_includes/resource-workflows.html                                                 |    82 |   46.03K |  0.078
topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.md                  |     1 |   48.63K |  0.076
_layouts/training_philosophies.html                                               |     1 |   35.49K |  0.065
_includes/help.html                                                               |   158 |  192.25K |  0.060
_layouts/introduction_slides.html                                                 |    11 |   44.96K |  0.060


                    done in 94.268 seconds.
```

**Comment by martenson** [comment]
may be remedied by https://github.com/galaxyproject/training-material/pull/1551

**Comment by martenson** [comment]
For comparison, this is a first build from heroku dyno. They are doing something smart, so they are often udner the 60s time, but not always.

```
2019-09-20T17:15:27.693986+00:00 heroku[web.1]: Starting process with command `bundle exec jekyll serve --baseurl "" -H 0.0.0.0 -P 44949`
2019-09-20T17:15:30.199065+00:00 app[web.1]: Configuration file: /app/_config.yml
2019-09-20T17:15:30.744176+00:00 app[web.1]:             Source: .
2019-09-20T17:15:30.744232+00:00 app[web.1]:        Destination: ./_site
2019-09-20T17:15:30.744237+00:00 app[web.1]:  Incremental build: disabled. Enable with --incremental
2019-09-20T17:15:30.744239+00:00 app[web.1]:       Generating... 
2019-09-20T17:15:31.684144+00:00 app[web.1]:        Jekyll Feed: Generating feed for posts
2019-09-20T17:16:22.305447+00:00 app[web.1]:                     done in 51.561 seconds.
2019-09-20T17:16:22.805611+00:00 app[web.1]:  Auto-regeneration: enabled for '.'
2019-09-20T17:16:22.908480+00:00 app[web.1]:     Server address: http://0.0.0.0:44949/
2019-09-20T17:16:22.908865+00:00 app[web.1]:   Server running... press ctrl-c to stop.
2019-09-20T17:16:23.452583+00:00 heroku[web.1]: State changed from starting to up
```

**Comment by erasche** [comment]
Will be fixed in #1551
